x/crosschain/keeper/grpc_query_cctx_rate_limit_test.go

```diff
@@ -123,60 +124,486 @@ func setupForeignCoins(
 	}
 }
 
-func TestKeeper_ListPendingCctxWithinRateLimit(t *testing.T) {
+func Test_ConvertCctxValue(t *testing.T) {
 	// chain IDs
 	ethChainID := getValidEthChainID()
+	btcChainID := getValidBtcChainID()
+
+	// zrc20 addresses for ETH, BTC, USDT and asset for USDT
+	zrc20ETH := sample.EthAddress().Hex()
+	zrc20BTC := sample.EthAddress().Hex()
+	zrc20USDT := sample.EthAddress().Hex()
+	assetUSDT := sample.EthAddress().Hex()
+
+	k, ctx, _, zk := keepertest.CrosschainKeeper(t)
+
+	// Set TSS
+	tss := sample.Tss()
+	zk.ObserverKeeper.SetTSS(ctx, tss)
+
+	// Set foreign coins
+	setupForeignCoins(t, ctx, zk, zrc20ETH, zrc20BTC, zrc20USDT, assetUSDT)
+
+	// Set rate limiter flags
+	rateLimiterFlags := createTestRateLimiterFlags(500, math.NewUint(5000), zrc20ETH, zrc20BTC, zrc20USDT, "2500", "50000", "0.8")
+	k.SetRateLimiterFlags(ctx, *rateLimiterFlags)
+
+	// get rate limiter rates
+	gasCoinRates, erc20CoinRates := k.GetRateLimiterRates(ctx)
+	foreignCoinMap := zk.FungibleKeeper.GetAllForeignCoinMap(ctx)
+
+	t.Run("should convert cctx ZETA value correctly", func(t *testing.T) {
+		// create cctx with 0.3 ZETA
+		cctx := sample.CrossChainTx(t, fmt.Sprintf("%d-%d", ethChainID, 1))
+		cctx.InboundTxParams.CoinType = coin.CoinType_Zeta
+		cctx.InboundTxParams.Asset = ""
+		cctx.GetCurrentOutTxParam().Amount = sdk.NewUint(3e17) // 0.3 ZETA
+
+		// convert cctx value
+		value := keeper.ConvertCctxValue(ethChainID, cctx, gasCoinRates, erc20CoinRates, foreignCoinMap)
+		require.Equal(t, sdk.MustNewDecFromStr("0.3"), value)
+	})
+	t.Run("should convert cctx ETH value correctly", func(t *testing.T) {
+		// create cctx with 0.003 ETH
+		cctx := sample.CrossChainTx(t, fmt.Sprintf("%d-%d", ethChainID, 1))
+		cctx.InboundTxParams.CoinType = coin.CoinType_Gas
+		cctx.InboundTxParams.Asset = ""
+		cctx.GetCurrentOutTxParam().Amount = sdk.NewUint(3e15) // 0.003 ETH
+
+		// convert cctx value
+		value := keeper.ConvertCctxValue(ethChainID, cctx, gasCoinRates, erc20CoinRates, foreignCoinMap)
+		require.Equal(t, sdk.MustNewDecFromStr("7.5"), value)
+	})
+	t.Run("should convert cctx BTC value correctly", func(t *testing.T) {
+		// create cctx with 0.0007 BTC
+		cctx := sample.CrossChainTx(t, fmt.Sprintf("%d-%d", btcChainID, 1))
+		cctx.InboundTxParams.CoinType = coin.CoinType_Gas
+		cctx.InboundTxParams.Asset = ""
+		cctx.GetCurrentOutTxParam().Amount = sdk.NewUint(70000) // 0.0007 BTC
+
+		// convert cctx value
+		value := keeper.ConvertCctxValue(btcChainID, cctx, gasCoinRates, erc20CoinRates, foreignCoinMap)
+		require.Equal(t, sdk.MustNewDecFromStr("35.0"), value)
```

Comment:
We can be ideal for tests is using values easy to calculate and update. 
Example: 0.0001 * 10000 = 1.0

I think this is fine here if we can precise in comments  35  = 0.0007 * 50000

And same for other

Reply:
https://github.com/zeta-chain/node/pull/2060/commits/578afccb54c86839c9c505dab05893132164b7ee